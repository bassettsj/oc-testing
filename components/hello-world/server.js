module.exports.data = (context, callback) => {
  callback(null, {
    name: context.params.name || 'John Doe'
  })
}
