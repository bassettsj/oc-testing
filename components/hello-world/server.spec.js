import nightmare from 'nightmare';

import 'babel-core/register';
import 'babel-polyfill';

import { data } from './server';

// unit
const scenarios = [
  { name: null },
  { name: undefined },
  { name: '' },
  { name: 'Jane Doe' },
  { name: 'Zlatan IbrahimoviÄ‡' }
];

scenarios.forEach((scenario) => {
  test(`when name is "${scenario.name}"`, () => {
    const context = { params: { name: scenario.name } };
    data(context, (error, model) => {
      expect(model).toMatchSnapshot();
    });
  });
});

// ui
const name = 'Superman';

test.skip(`Hello ${name}`, async () => {
  const page = nightmare().goto(`http://localhost:3030/hello-world/?name=${name}`);
  const text = await page.evaluate(() => document.body.textContent).end();
  expect(text).toContain(`Hello ${name}`);
  expect(text).toMatchSnapshot();
});
