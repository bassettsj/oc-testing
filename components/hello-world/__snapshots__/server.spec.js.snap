// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Superman 1`] = `"{\\"type\\":\\"oc-component-local\\",\\"version\\":\\"1.0.0\\",\\"requestVersion\\":\\"\\",\\"name\\":\\"hello-world\\",\\"renderMode\\":\\"rendered\\",\\"href\\":\\"http://localhost:3030/hello-world?name=Superman\\",\\"html\\":\\"<div>Hello Superman</div>\\"}"`;

exports[`when name is "" 1`] = `
Object {
  "name": "John Doe",
}
`;

exports[`when name is "Jane Doe" 1`] = `
Object {
  "name": "Jane Doe",
}
`;

exports[`when name is "Zlatan Ibrahimović" 1`] = `
Object {
  "name": "Zlatan Ibrahimović",
}
`;

exports[`when name is "null" 1`] = `
Object {
  "name": "John Doe",
}
`;

exports[`when name is "undefined" 1`] = `
Object {
  "name": "John Doe",
}
`;
